RSSDP/SsdpDevicePublisher.cs

```diff
@@ -250,7 +257,15 @@ private void ProcessSearchRequest(string mx, string searchTarget, IpEndPointInfo
 
                     foreach (var device in deviceList)
                     {
-                        SendDeviceSearchResponses(device, remoteEndPoint, receivedOnlocalIpAddress, cancellationToken);
+                        var matched = true;
+                        if (_sendOnlyMatchedHost && !_networkManager.IsInSameSubnet(device.ToRootDevice().Address, remoteEndPoint.IpAddress, device.ToRootDevice().SubnetMask))
+                        {
+                            matched = false;
+                        }
+                        if (matched)
```

Comment:
umm... why not just directly check for condition instead of keeping a `matched` state?
```suggestion
                        if (!_sendOnlyMatchedHost ||
                            _networkManager.IsInSameSubnet(device.ToRootDevice().Address, remoteEndPoint.IpAddress, device.ToRootDevice().SubnetMask))
```